lib/validate_url.rb

```diff
@@ -13,28 +13,32 @@ def initialize(options)
         options.reverse_merge!(message: :url)
         options.reverse_merge!(no_local: false)
         options.reverse_merge!(public_suffix: false)
+        options.reverse_merge!(accept_array: false)
 
         super(options)
       end
 
       def validate_each(record, attribute, value)
         schemes = [*options.fetch(:schemes)].map(&:to_s)
-        begin
-          uri = URI.parse(value)
-          host = uri && uri.host
-          scheme = uri && uri.scheme
+        if value.respond_to?(:each)
+          # Error out if we're not allowing arrays
+          if !options.include?(:accept_array) || !options.fetch(:accept_array)
+            record.errors.add(attribute, :url, **filtered_options(value))
+          end
 
-          valid_raw_url = scheme && value =~ /\A#{URI::regexp([scheme])}\z/
-          valid_scheme = host && scheme && schemes.include?(scheme)
-          valid_no_local = !options.fetch(:no_local) || (host && host.include?('.'))
-          valid_suffix = !options.fetch(:public_suffix) || (host && PublicSuffix.valid?(host, :default_rule => nil))
+          # We have to manually handle `:allow_nil` and `:allow_blank` since it's not caught by
+          # ActiveRecord's own validators. We do that by just removing all the nil's if we want to
+          # allow them so it's not passed on later.
+          value = value.reject(&:nil?) if options.include?(:allow_nil) && options.fetch(:allow_nil)
+          value = value.reject(&:blank?) if options.include?(:allow_blank) && options.fetch(:allow_blank)
 
-          unless valid_raw_url && valid_scheme && valid_no_local && valid_suffix
-            record.errors.add(attribute, options.fetch(:message), value: value)
-          end
-        rescue URI::InvalidURIError
-          record.errors.add(attribute, :url, **filtered_options(value))
+          result = value.map { |v| validate_url(record, attribute, v, schemes) }
```

Comment:
use flat_map instead, please

Reply:
Good catch, this will be fixed in the next push.